sdk/core/core-tracing/src/tracingClient.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  TracingClient,
+  OperationTracingOptions,
+  TracingSpan,
+  TracingContext,
+  TracingSpanOptions,
+  TracingSpanContext,
+  TracingClientOptions,
+  Instrumenter
+} from "./interfaces";
+import { getInstrumenter } from "./instrumenter";
+import { knownContextKeys } from "./tracingContext";
+
+/** @internal */
+export class TracingClientImpl implements TracingClient {
+  private _namespace: string;
+  private _packageName: string;
+  private _packageVersion?: string;
+
+  constructor(options: TracingClientOptions) {
+    this._namespace = options.namespace;
+    this._packageName = options.packageName;
+    this._packageVersion = options.packageVersion;
+  }
+  startSpan<Options extends { tracingOptions?: OperationTracingOptions }>(
+    name: string,
+    operationOptions?: Options,
+    spanOptions?: TracingSpanOptions
+  ): {
+    span: TracingSpan;
+    tracingContext: TracingContext;
+    updatedOptions: Options;
+  } {
+    const startSpanResult = this.getInstrumenter().startSpan(name, {
+      ...spanOptions,
+      packageName: this._packageName,
+      packageVersion: this._packageVersion,
+      tracingContext: operationOptions?.tracingOptions?.tracingContext
+    });
+    let tracingContext = startSpanResult.tracingContext;
+    const span = startSpanResult.span;
+    if (!tracingContext.getValue(knownContextKeys.Namespace)) {
+      tracingContext = tracingContext.setValue(knownContextKeys.Namespace, this._namespace);
+    }
+    span.setAttribute("az.namespace", tracingContext.getValue(knownContextKeys.Namespace));
+    const updatedOptions = {
+      ...operationOptions,
+      tracingOptions: {
+        tracingContext: tracingContext
+      }
+    } as Options;
+    return {
+      span,
+      tracingContext: tracingContext,
+      updatedOptions
```

Comment:
Consider having the new context returned only in the updatedOptions bag instead of two places...

Can we omit tracingOptions entirely from updatedOptions?